module/zfs/bqueue.c

```diff
@@ -87,13 +87,17 @@ bqueue_enqueue(bqueue_t *q, void *data, uint64_t item_size)
 void *
 bqueue_dequeue(bqueue_t *q)
 {
-	void *ret;
+	void *ret = NULL;
 	uint64_t item_size;
 	mutex_enter(&q->bq_lock);
 	while (q->bq_size == 0) {
 		cv_wait(&q->bq_pop_cv, &q->bq_lock);
 	}
 	ret = list_remove_head(&q->bq_list);
+	if (ret == NULL) {
```

Comment:
That seems fine. But why is ret being initialized to NULL? That doesn't seem to accomplish anything, given that ret is unconditionally set by a call to list_remove_head().